plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -370,6 +372,17 @@ public Transaction newCreateTableTransaction(
                 Optional.of(session.getUser()));
     }
 
+    @Override
+    public void registerTable(ConnectorSession session, SchemaTableName schemaTableName, String tableLocation, String metadataLocation)
+            throws TrinoException
+    {
+        TableInput tableInput = getTableInput(schemaTableName.getTableName(), Optional.of(session.getUser()), ImmutableMap.<String, String>builder()
+                .put(TABLE_TYPE_PROP, ICEBERG_TABLE_TYPE_VALUE.toUpperCase(ENGLISH))
```

Comment:
Follow up, but we can probably move this line to `getTableInput`

Reply:
If we move the `TABLE_TYPE_PROP` property to `getTableInput` method then the caller of this method needs to ensure that `parameters` arg should not contain `TABLE_TYPE_PROP`.

One of the places, where the caller needs to remove the  `TABLE_TYPE_PROP` from table parameters before calling `getTableInput`.

https://github.com/trinodb/trino/blob/5d124c47dd134f0026512ec8c8e3290ed086c8d6/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestIcebergGlueCatalogSkipArchive.java#L114-L115

Reply:
You could have a set-if-not-present kind of thing

Reply:
Created an issue for this https://github.com/trinodb/trino/issues/14967